client/state/sites/plans/assembler.js

```diff
@@ -17,6 +17,7 @@ export const createSitePlanObject = ( plan ) => {
 		formattedPrice: plan.formatted_price,
 		freeTrial: Boolean( plan.free_trial ),
 		hasDomainCredit: Boolean( plan.has_domain_credit ),
+		hasRedeemedDomainCredit: Boolean( plan.has_redeemed_domain_credit ),
```

Comment:
We probably shouldn't be extending this with more properties. The plan is to deprecate this state slice completely in favor of the plans data-store (see https://github.com/Automattic/wp-calypso/issues/86638). I would consider adding this to the return value from [use-site-plans](https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/plans/queries/use-site-plans.ts) and use `useCurrentPlan` hook from there to bring it into `plans-features-main`.

Reply:
Thanks for the feedback Christos -- exploring this now 🕵️

Reply:
[0990b75](https://github.com/Automattic/wp-calypso/pull/89060/commits/0990b75fd673e06fb0daa9e5a42553cb86ee3c28)